wire-ios/Wire-iOS/Sources/AppRootRouter.swift

```diff
@@ -367,6 +372,14 @@ extension AppRootRouter {
                                completion: completion)
     }
 
+    private func showRecovery(completion: @escaping () -> Void) {
+        guard let launchOptions = lastLaunchOptions else { return }
+        completion()
+        enqueueTransition(to: .headless) { [weak self] in
+            self?.sessionManager.start(launchOptions: launchOptions)
+        }
+    }
```

Comment:
what does this do exactly?

Reply:
Basically it re-triggers the steps that are performed while starting a session (trying to migrate database) just as the first time.
It turns out that retrying was not possible at all as it was dependent on launchOptions and launching process.